Packages/com.unity.inputsystem/InputSystem/State/InputState.cs

```diff
@@ -122,12 +122,20 @@ public static unsafe void Change(InputDevice device, InputEventPtr eventPtr, Inp
             var statePtr = UnsafeUtility.AddressOf(ref state);
             var stateOffset = control.stateBlock.byteOffset - device.stateBlock.byteOffset;
 
+            // Report any calls with no event and no specified time (relying on both default parameter values)
+            // This may not matter but would like to know about these during testing with more devices
+            //
+            if (!eventPtr.valid && time == 0.0)
```

Comment:
Should we still proceed without return if this scenario occurs?

Reply:
Good question - I thought it better to continue rather than just return here.

I haven't actually seen this case happen yet with my limited devices for testing.

What might be better is to simply set the time here in this case:

`            if (!eventPtr.valid && time == default)
            {
                Debug.Log($"InputState.Change<>('{control.name}') called with null event and default time\n");
                time = InputRuntime.s_Instance.currentTime;
            }
`
I think this would preserve the existing behaviour where it would query the time if no event was provided.

Reply:
I've refactored this so we no longer need this check.

There are two paths into this function from the two **Change<>()** templated functions.

The original **Change<>()** function always passes in an event (with it's own internal time) - if the event is not valid it passes the current time.

The other newly added **Change<>()** function always passes in a time value (it's a required parameter) along with a dummy event - this is the one used by the **FastMouse.OnNextUpdate()** function.